src/dbnode/client/session.go

```diff
@@ -1185,7 +1185,7 @@ func (s *session) fetchTaggedAttemptWithRLock(
 ) (*fetchState, error) {
 	// NB(prateek): we have to clone the namespace, as we cannot guarantee the lifecycle
 	// of the hostQueues responding is less than the lifecycle of the current method.
-	nsClone := s.pools.id.Clone(ns)
+	nsClone := s.maybeClone(ns)
```

Comment:
lets skip the fetch path for now. the impact isn't going to be nearly as pronounced and i don't want to land it without a lot more testing.